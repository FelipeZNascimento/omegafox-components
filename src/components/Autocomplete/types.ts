export type TDropdownItem = {
  id: number;
  name: string;
  abbreviationEn: string;
};

export interface IAutocompleteProps {
  dropdownList: TDropdownItem[];
  onSelect: (item: TDropdownItem) => void;
}
