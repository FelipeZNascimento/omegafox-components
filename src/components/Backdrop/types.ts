export interface IBackdropProps {
  align?: 'left' | 'center' | 'right';
  children: JSX.Element;
  isOpen: boolean;
}
