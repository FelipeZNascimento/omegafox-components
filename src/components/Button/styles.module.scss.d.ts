// This file is automatically generated.
// Please do not change this file!
interface CssExports {
  'bottom-margin-l': string;
  'bottom-margin-m': string;
  'bottom-margin-none': string;
  'bottom-margin-s': string;
  'bottom-margin-xl': string;
  'bottom-margin-xs': string;
  'bottom-margin-xxl': string;
  'bottom-margin-xxs': string;
  'bottom-padding-l1': string;
  'bottom-padding-l2': string;
  'bottom-padding-l3': string;
  'bottom-padding-l4': string;
  'bottom-padding-m': string;
  'bottom-padding-none': string;
  'bottom-padding-s1': string;
  'bottom-padding-s2': string;
  'bottom-padding-s3': string;
  'children': string;
  'color-black': string;
  'color-black-bg': string;
  'color-black-to-black': string;
  'color-black-to-white': string;
  'color-blue': string;
  'color-blue-bg': string;
  'color-blue-to-black': string;
  'color-blue-to-white': string;
  'color-gold': string;
  'color-gold-bg': string;
  'color-gold-to-black': string;
  'color-gold-to-white': string;
  'color-green': string;
  'color-green-bg': string;
  'color-green-to-black': string;
  'color-green-to-white': string;
  'color-grey1': string;
  'color-grey1-bg': string;
  'color-grey1-to-black': string;
  'color-grey1-to-white': string;
  'color-grey2': string;
  'color-grey2-bg': string;
  'color-grey2-to-black': string;
  'color-grey2-to-white': string;
  'color-grey3': string;
  'color-grey3-bg': string;
  'color-grey3-to-black': string;
  'color-grey3-to-white': string;
  'color-grey4': string;
  'color-grey4-bg': string;
  'color-grey4-to-black': string;
  'color-grey4-to-white': string;
  'color-grey5': string;
  'color-grey5-bg': string;
  'color-grey5-to-black': string;
  'color-grey5-to-white': string;
  'color-link': string;
  'color-link-bg': string;
  'color-link-to-black': string;
  'color-link-to-white': string;
  'color-mint': string;
  'color-mint-bg': string;
  'color-mint-to-black': string;
  'color-mint-to-white': string;
  'color-navy': string;
  'color-navy-bg': string;
  'color-navy-to-black': string;
  'color-navy-to-white': string;
  'color-orange': string;
  'color-orange-bg': string;
  'color-orange-to-black': string;
  'color-orange-to-white': string;
  'color-purple': string;
  'color-purple-bg': string;
  'color-purple-to-black': string;
  'color-purple-to-white': string;
  'color-red': string;
  'color-red-bg': string;
  'color-red-to-black': string;
  'color-red-to-white': string;
  'color-sky': string;
  'color-sky-bg': string;
  'color-sky-to-black': string;
  'color-sky-to-white': string;
  'color-white': string;
  'color-white-bg': string;
  'color-white-to-black': string;
  'color-white-to-white': string;
  'confirm': string;
  'container': string;
  'containerLarge': string;
  'containerMedium': string;
  'containerSmall': string;
  'danger': string;
  'disabled': string;
  'drop-shadowed': string;
  'enabled': string;
  'enabledDark': string;
  'enabledLight': string;
  'font-size-l': string;
  'font-size-m': string;
  'font-size-m2': string;
  'font-size-s': string;
  'font-size-xl': string;
  'font-size-xs': string;
  'font-size-xxl': string;
  'font-size-xxs': string;
  'icon': string;
  'iconLeft': string;
  'iconRight': string;
  'left-margin-l': string;
  'left-margin-m': string;
  'left-margin-none': string;
  'left-margin-s': string;
  'left-margin-xl': string;
  'left-margin-xs': string;
  'left-margin-xxl': string;
  'left-margin-xxs': string;
  'left-padding-l1': string;
  'left-padding-l2': string;
  'left-padding-l3': string;
  'left-padding-l4': string;
  'left-padding-m': string;
  'left-padding-none': string;
  'left-padding-s1': string;
  'left-padding-s2': string;
  'left-padding-s3': string;
  'margin-l': string;
  'margin-m': string;
  'margin-none': string;
  'margin-s': string;
  'margin-xl': string;
  'margin-xs': string;
  'margin-xxl': string;
  'margin-xxs': string;
  'neutral': string;
  'padding-l1': string;
  'padding-l2': string;
  'padding-l3': string;
  'padding-l4': string;
  'padding-m': string;
  'padding-none': string;
  'padding-s1': string;
  'padding-s2': string;
  'padding-s3': string;
  'primary': string;
  'right-margin-l': string;
  'right-margin-m': string;
  'right-margin-none': string;
  'right-margin-s': string;
  'right-margin-xl': string;
  'right-margin-xs': string;
  'right-margin-xxl': string;
  'right-margin-xxs': string;
  'right-padding-l1': string;
  'right-padding-l2': string;
  'right-padding-l3': string;
  'right-padding-l4': string;
  'right-padding-m': string;
  'right-padding-none': string;
  'right-padding-s1': string;
  'right-padding-s2': string;
  'right-padding-s3': string;
  'rounded': string;
  'selected': string;
  'shadowed': string;
  'top-margin-l': string;
  'top-margin-m': string;
  'top-margin-none': string;
  'top-margin-s': string;
  'top-margin-xl': string;
  'top-margin-xs': string;
  'top-margin-xxl': string;
  'top-margin-xxs': string;
  'top-padding-l1': string;
  'top-padding-l2': string;
  'top-padding-l3': string;
  'top-padding-l4': string;
  'top-padding-m': string;
  'top-padding-none': string;
  'top-padding-s1': string;
  'top-padding-s2': string;
  'top-padding-s3': string;
  'warning': string;
}
declare var cssExports: CssExports;
export = cssExports;
