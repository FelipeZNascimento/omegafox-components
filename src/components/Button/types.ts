export interface IButtonProps extends React.HTMLAttributes<HTMLButtonElement> {
  children?: React.ReactNode | null;
  icon?: React.ReactNode | null;
  isDisabled?: boolean;
  isDarkMode?: boolean;
  isSelected?: boolean;
  leftIcon?: boolean;
  size?: 'small' | 'medium' | 'large';
  shape?: 'rounded' | 'squared';
  shadowed?: boolean;
  tooltip?: string;
  variant?: 'primary' | 'danger' | 'neutral' | 'confirm' | 'warning';
  onClick: () => void;
}
