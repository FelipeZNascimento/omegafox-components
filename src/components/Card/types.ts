export interface ICardProps {
  isForceMobile?: boolean;
  isSelected: boolean;
  title: string;
  subtitle?: string | null;
  onClick?: null | (() => void);
  renderingStatusFunction?: null | (() => JSX.Element | null);
}
