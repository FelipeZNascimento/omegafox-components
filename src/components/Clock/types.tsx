import {
  FOOTBALL_MATCH_STATUS,
  AMERICAN_FOOTBALL_MATCH_STATUS
} from './constants';
import { TBetValues } from '../types';

type TClock = {
  time: number;
};

type FootballKeys = keyof typeof FOOTBALL_MATCH_STATUS;
export type TClockFootball = TClock & {
  status: typeof FOOTBALL_MATCH_STATUS[FootballKeys];
};

type AmericanFootballKeys = keyof typeof AMERICAN_FOOTBALL_MATCH_STATUS;
export type TClockAmericanFootball = TClock & {
  status: typeof AMERICAN_FOOTBALL_MATCH_STATUS[AmericanFootballKeys];
};

export interface IClockProps {
  betValue: TBetValues | null;
  clock: TClockAmericanFootball | TClockFootball;
  isMatchStarted?: boolean;
  isExpandable: boolean;
  isExpanded: boolean;
  isMatchEditable: boolean;
  timestamp: number;
}

export interface IClockMobileProps {
  clock: TClockAmericanFootball | TClockFootball;
  timestamp: number;
}
