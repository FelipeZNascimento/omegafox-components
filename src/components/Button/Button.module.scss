@import "../../styling/alignments.scss";
@import "../../styling/colors.scss";
@import "../../styling/font-sizes.scss";
@import "../../styling/margins.scss";
@import "../../styling/paddings.scss";

.container {
  @include flex-aligned(center, center);

  position: relative;
  text-transform: uppercase;
  width: 100%;

  &Small {
    @include font-sized(m);
    @include padded(horizontal, m);
    @include padded(vertical, s1);
  }

  &Medium {
    @include font-sized(m);
    @include padded(horizontal, l1);
    @include padded(vertical, m);
  }

  &Large {
    @include font-sized(l);
    @include padded(horizontal, l3);
    @include padded(vertical, l1);
  }
}

.enabled {
  cursor: pointer;
  filter: brightness(90%);
  transition: all 0.3s ease-in-out;

  &:hover {
    filter: brightness(100%);
  }

  &:after {
    content: "";
    display: block;
    position: absolute;
    left: 0;
    top: 0;
    width: 100%;
    height: 100%;
    opacity: 0;
    transition: all 0.5s;
    box-shadow: 0 0 10px 10px black;
  }

  &:active:after {
    box-shadow: 0 0 0 0 black;
    position: absolute;
    left: 0;
    top: 0;
    opacity: 1;
    transition: 0s;
  }

  &Dark {
    @extend .enabled;
    &:after {
      box-shadow: 0 0 10px 10px $color-grey2;
    }
    &:active:after {
      box-shadow: 0 0 0 0 $color-grey2;
    }
  }

  &Light {
    @extend .enabled;
    &:after {
      box-shadow: 0 0 10px 10px $color-grey5;
    }
    &:active:after {
      box-shadow: 0 0 0 0 $color-grey5;
    }
  }
}

.selected {
  filter: brightness(100%);
}

.disabled {
  filter: brightness(50%);
  cursor: not-allowed;

  &:hover {
    filter: brightness(50%);
  }
}

.rounded {
  border-radius: 4px;

  &:after {
    border-radius: 4px;
  }
  &:active:after {
    border-radius: 4px;
  }
}

.shadowed {
  box-shadow: 0 5px 15px rgba(0, 0, 0, 0.3);
}

.children {
  @include flex-aligned(center, center);
}

.icon {
  @include flex-aligned(center, center);

  &Left {
    @include padded(right, m);
  }

  &Right {
    @include padded(left, m);
  }
}

.primary {
  @include bg-colored($color-blue);
  @include font-colored($color-white);

  border: none;
  // &:hover {
  //   background-color: $color-blue;
  // }
}

.danger {
  @include bg-colored($color-red);
  @include font-colored($color-white);

  border: none;
  // &:hover {
  //   background-color: $color-red;
  // }
}

.warning {
  @include bg-colored($color-orange);
  @include font-colored($color-white);

  border: none;
  // &:hover {
  //   background-color: $color-orange;
  // }
}

.confirm {
  @include bg-colored($color-green);
  @include font-colored($color-white);

  border: none;
  // &:hover {
  //   @include bg-colored($color-green);
  // }
}

.neutral {
  @include bg-colored($color-grey3);
  @include font-colored($color-white);
  border: none;

  // &:hover {
  //   @include bg-colored($color-grey3);
  // }
}
