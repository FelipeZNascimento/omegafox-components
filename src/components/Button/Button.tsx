import React from 'react';

import { IButtonProps } from './types';
import classNames from 'classnames';
import styles from './Button.module.scss';

export const Button = ({
  children = null,
  icon = null,
  isDarkMode = false,
  isDisabled = false,
  isSelected = false,
  leftIcon = true,
  size = 'medium',
  shadowed = true,
  shape = 'rounded',
  variant = 'primary',
  onClick,
  ...props
}: IButtonProps) => {
  const buttonClass = classNames(styles.container, {
    [styles.primary]: variant === 'primary',
    [styles.danger]: variant === 'danger',
    [styles.warning]: variant === 'warning',
    [styles.neutral]: variant === 'neutral',
    [styles.confirm]: variant === 'confirm',
    [styles.containerSmall]: size === 'small',
    [styles.containerMedium]: size === 'medium',
    [styles.containerLarge]: size === 'large',
    [styles.rounded]: shape === 'rounded',
    [styles.shadowed]: shadowed && !isDisabled,
    [styles.selected]: isSelected,
    [styles.enabledLight]: !isDisabled && !isDarkMode,
    [styles.enabledDark]: !isDisabled && isDarkMode,
    [styles.disabled]: isDisabled
  });

  const renderIcon = () => {
    if (!icon) {
      return;
    }

    const iconClass = classNames(styles.icon, {
      [styles.iconLeft]: children && leftIcon,
      [styles.iconRight]: children && !leftIcon
    });

    const testId = `test-button__icon-${leftIcon ? 'left' : 'right'}`;
    return (
      <div data-testid={testId} className={iconClass}>
        {icon}
      </div>
    );
  };

  const renderChildren = () => {
    return (
      <div data-testid="test-button__children" className={styles.children}>
        {children}
      </div>
    );
  };

  const checkOnClick = () => {
    if (!isDisabled) {
      onClick();
    }
  };

  return (
    <button
      data-testid="test-button"
      className={buttonClass}
      onClick={checkOnClick}
      {...props}
    >
      {icon && leftIcon && renderIcon()}
      {children && renderChildren()}
      {icon && !leftIcon && renderIcon()}
    </button>
  );
};
