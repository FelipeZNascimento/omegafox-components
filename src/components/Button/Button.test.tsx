import React from 'react';
import { fireEvent, render } from '@testing-library/react';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faChevronLeft } from '@fortawesome/free-solid-svg-icons';

import { Button } from './Button';
import { ButtonProps } from './types';

describe('ButtonComponent', () => {
  const renderComponent = ({
    children,
    icon,
    isDarkMode,
    isDisabled,
    isSelected,
    leftIcon,
    size,
    shadowed,
    shape,
    tooltip,
    variant
  }: Partial<ButtonProps>) =>
    render(
      <Button
        icon={icon}
        isDarkMode={isDarkMode}
        isDisabled={isDisabled}
        isSelected={isSelected}
        leftIcon={leftIcon}
        size={size}
        shadowed={shadowed}
        shape={shape}
        tooltip={tooltip}
        variant={variant}
        onClick={jest.fn}
      >
        {children}
      </Button>
    );

  it('should render', () => {
    const buttonText = 'Click';
    const { container } = renderComponent({ children: buttonText });

    expect(container).toMatchSnapshot();
  });

  it('should render correctly with default values', () => {
    const buttonText = 'Click';
    const { getByTestId, queryByTestId } = renderComponent({
      children: buttonText
    });

    const button = getByTestId('test-button');
    const buttonChildren = getByTestId('test-button__children');
    const buttonIcon = queryByTestId('test-button__icon-left');

    expect(button).toHaveClass('shadowed');
    expect(button).toHaveClass('rounded');
    expect(button).toHaveClass('enabledLight');
    expect(button).toHaveClass('containerMedium');
    expect(button).toHaveClass('primary');

    expect(buttonIcon).toBeNull();
    expect(buttonChildren).toHaveTextContent(buttonText);
  });

  it('should render left icon correctly', () => {
    const buttonText = 'Click';
    const { getByTestId } = renderComponent({
      children: buttonText,
      leftIcon: true,
      icon: <FontAwesomeIcon icon={faChevronLeft} size="lg" />
    });
    const buttonIcon = getByTestId('test-button__icon-left');

    expect(buttonIcon).toBeVisible();
    expect(buttonIcon).toHaveClass('iconLeft');
  });

  it('should render right icon correctly', () => {
    const buttonText = 'Click';
    const { getByTestId } = renderComponent({
      children: buttonText,
      leftIcon: false,
      icon: <FontAwesomeIcon icon={faChevronLeft} size="lg" />
    });
    const buttonIcon = getByTestId('test-button__icon-right');

    expect(buttonIcon).toBeVisible();
    expect(buttonIcon).toHaveClass('iconRight');
  });

  it('should callback on click', () => {
    const onClickFn = jest.fn();
    const buttonText = 'Click';

    const { getByTestId } = render(
      <Button onClick={onClickFn}>{buttonText}</Button>
    );

    fireEvent.click(getByTestId('test-button'));
    expect(onClickFn).toHaveBeenCalled();
  });
});
