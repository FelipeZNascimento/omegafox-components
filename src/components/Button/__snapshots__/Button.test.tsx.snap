// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonComponent should render 1`] = `
<div>
  <button
    class="container primary containerMedium rounded shadowed enabledLight"
    data-testid="test-button"
  >
    <div
      class="children"
      data-testid="test-button__children"
    >
      Click
    </div>
  </button>
</div>
`;
